File: resource-registry/src/test/java/nl/haploid/octowight/registry/data/ResourceRootFactoryTest.java
package nl.haploid.octowight.registry.data;

import mockit.Tested;
import nl.haploid.octowight.AtomChangeEvent;
import nl.haploid.octowight.registry.TestData;
import nl.haploid.octowight.registry.repository.ResourceDmo;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class ResourceRootFactoryTest {

	@Tested
	private ResourceRootFactory resourceRootFactory;

	@Test
	public void testFromAtomChangeEvent() {
		final AtomChangeEvent event = TestData.atomChangeEvent("pam");
		final String resourceType = "greer";
		final ResourceRoot resourceRoot = resourceRootFactory.fromAtomChangeEvent(event, resourceType);
		assertEquals(event.getAtomId(), resourceRoot.getAtomId());
		assertEquals(event.getAtomLocus(), resourceRoot.getAtomLocus());
		assertEquals(event.getAtomType(), resourceRoot.getAtomType());
		assertNull(resourceRoot.getResourceId());
		assertEquals(resourceType, resourceRoot.getResourceType());
	}

	@Test
	public void testFromResourceDmo() {
		final ResourceDmo resourceDmo = TestData.resourceDmo();
		final ResourceRoot resourceRoot = resourceRootFactory.fromResourceDmo(resourceDmo);
		assertEquals(resourceDmo.getAtomId(), resourceRoot.getAtomId());
		assertEquals(resourceDmo.getAtomType(), resourceRoot.getAtomType());
		assertEquals(resourceDmo.getAtomLocus(), resourceRoot.getAtomLocus());
		assertEquals(resourceDmo.getResourceId(), resourceRoot.getResourceId());
		assertEquals(resourceDmo.getResourceType(), resourceRoot.getResourceType());
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/data/ResourceRootFactoryTest.scala
package nl.haploid.octowight.registry.data

import nl.haploid.octowight.Tested
import nl.haploid.octowight.registry.{AbstractTest, TestData}

class ResourceRootFactoryTest extends AbstractTest {
  @Tested private val resourceRootFactory: ResourceRootFactory = null

  "Resource root factory" should "build from event & resource type" in {
    val event = TestData.atomChangeEvent("pam")
    val resourceType = "greer"
    val resourceRoot = resourceRootFactory.fromAtomChangeEvent(event, resourceType)
    resourceRoot.getAtomId should be(event.getAtomId)
    resourceRoot.getAtomOrigin should be(event.getAtomOrigin)
    resourceRoot.getAtomType should be(event.getAtomType)
    resourceRoot.getResourceId should be(null)
    resourceRoot.getResourceType should be(resourceType)
  }

  "Resource root factory" should "build from resource root DMO" in {
    val resourceRootDmo = TestData.resourceRootDmo
    val resourceRoot = resourceRootFactory.fromResourceRootDmo(resourceRootDmo)
    resourceRoot.getAtomId should be(resourceRootDmo.getAtomId)
    resourceRoot.getAtomType should be(resourceRootDmo.getAtomType)
    resourceRoot.getAtomOrigin should be(resourceRootDmo.getAtomOrigin)
    resourceRoot.getResourceId should be(resourceRootDmo.getResourceId)
    resourceRoot.getResourceType should be(resourceRootDmo.getResourceType)
  }
}
